rest/admin_api_test.go

```diff
@@ -3001,3 +3002,184 @@ func TestChannelNameSizeWarningDeleteChannel(t *testing.T) {
 		assert.Equal(t, before, after)
 	})
 }
+
+func TestConfigEndpoint(t *testing.T) {
+	testCases := []struct {
+		Name              string
+		Config            string
+		ConsoleLevel      base.LogLevel
+		ConsoleLogKeys    []string
+		ExpectError       bool
+		FileLoggerCheckFn func() bool
+	}{
+		{
+			Name: "Set LogLevel and LogKeys",
+			Config: `
+			{
+				"logging": {
+					"console": {
+						"log_level": "trace",
+						"log_keys": ["Config"]
+					}
+				}
+			}`,
+			ConsoleLevel:   base.LevelTrace,
+			ConsoleLogKeys: []string{"Config"},
+			ExpectError:    false,
+		},
+		{
+			Name: "Set LogLevel and multiple LogKeys",
+			Config: `
+			{
+				"logging": {
+					"console": {
+						"log_level": "info",
+						"log_keys": ["Config", "HTTP+"]
+					}
+				}
+			}`,
+			ConsoleLevel:   base.LevelInfo,
+			ConsoleLogKeys: []string{"Config", "HTTP"},
+			ExpectError:    false,
+		},
+		{
+			Name: "Set Invalid Fields",
+			Config: `
+			{
+				"logging": {
+					"console": {
+						"log_level": "info",
+						"log_keys": ["Config", "HTTP+"]
+					},
+					"fake": {}
```

Comment:
Would prefer to set a non-runtime configurable field instead of an unknown field here. Feel free to add as an additional test-case instead though.

```suggestion
			Name: "Set non-configurable field",
			Config: `
			{
				"logging": {
					"console": {
						"log_level": "info",
						"log_keys": ["Config", "HTTP+"]
					}
				},
				"bootstrap": {
					"server": "couchbase://203.0.113.135"
				}
```